src/local_device.rs

```diff
@@ -275,3 +275,142 @@ impl LocalDevice {
+    pub fn new(py_kwargs: Option<&PyDict>) -> PyResult<Self> {
+        crate::binding_utils::parse_kwargs!(
+            py_kwargs,
+            [user_id: UserID, "user_id"],
+            [human_handle: Option<HumanHandle>, "human_handle"],
+            [profile, "profile", py_to_rs_user_profile],
+            [created_on: DateTime, "created_on"],
+            [revoked_on: Option<DateTime>, "revoked_on"],
+        );
+
+        Ok(Self(libparsec::client_types::UserInfo {
+            user_id: user_id.0,
+            human_handle: human_handle.map(|x| x.0),
+            profile,
+            created_on: created_on.0,
+            revoked_on: revoked_on.map(|x| x.0),
+        }))
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.0))
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> bool {
+        match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            _ => unimplemented!(),
+        }
+    }
+
+    #[getter]
+    fn user_id(&self) -> PyResult<UserID> {
+        Ok(UserID(self.0.user_id.clone()))
+    }
+
+    #[getter]
+    fn human_handle(&self) -> PyResult<Option<HumanHandle>> {
+        Ok(self.0.human_handle.as_ref().map(|x| HumanHandle(x.clone())))
+    }
+
+    #[getter]
+    fn profile(&self) -> PyResult<PyObject> {
+        rs_to_py_user_profile(&self.0.profile)
+    }
+
+    #[getter]
+    fn created_on(&self) -> PyResult<DateTime> {
+        Ok(DateTime(self.0.created_on))
+    }
+
+    #[getter]
+    fn revoked_on(&self) -> PyResult<Option<DateTime>> {
+        Ok(self.0.revoked_on.map(DateTime))
+    }
+
+    fn user_display(&self) -> PyResult<String> {
+        Ok(self.0.user_display())
+    }
+
+    fn short_user_display(&self) -> PyResult<String> {
+        Ok(self.0.short_user_display())
+    }
+
+    fn is_revoked(&self) -> PyResult<bool> {
+        Ok(self.0.is_revoked())
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct DeviceInfo(pub libparsec::client_types::DeviceInfo);
+
+#[pymethods]
+impl DeviceInfo {
+    #[new]
+    #[args(py_kwargs = "**")]
+    pub fn new(py_kwargs: Option<&PyDict>) -> PyResult<Self> {
+        crate::binding_utils::parse_kwargs!(
+            py_kwargs,
+            [device_id: DeviceID, "device_id"],
+            [device_label: Option<DeviceLabel>, "device_label"],
+            [created_on: DateTime, "created_on"],
+        );
+
+        Ok(Self(libparsec::client_types::DeviceInfo {
+            device_id: device_id.0,
+            device_label: device_label.map(|x| x.0),
+            created_on: created_on.0,
+        }))
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.0))
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> bool {
+        match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            CompareOp::Lt => self.0 < other.0,
```

Comment:
Why do we need the full range of comparison here ? (given we only have eq/ne for UserInfo)

Reply:
There is a sort in tests

Reply:
Then I think we should also implement this for the UserInfo (given it should be trivial do to and it's probably interesting to have deterministic order when testing/displaying)